bridges/snowbridge/pallets/system/src/lib.rs

```diff
@@ -617,12 +642,12 @@ pub mod pallet {
 			location: Box<VersionedLocation>,
 			metadata: AssetMetadata,
 		) -> DispatchResult {
-			ensure_root(origin)?;
+			let who = T::RegisterTokenOrigin::ensure_origin(origin)?;
```

Comment:
I've realized that if we make token registration permissionless, the origin should be an XCM location (i.e Transact.OriginKind = Xcm), and we should verify that the origin location is a parent of the token location.

Otherwise any random user can register a token with incorrect or misleading metadata

Reply:
How would you register a token like KILT? 
1. By sending an XCM directly from KILT parachain? That would require they setup an HRMP channel with bridge hub just for registration. 
2. Impersonating KILT using governance from the relaychain.

Reply:
Yeah good point - another idea I had was to move the registration extrinsic to AH which of course can read the asset metadata directly from storage.

Reply:
but for now, yes, we should with root